### PR TITLE
[spark] Add rollback ddl syntax

### PR DESCRIPTION
**Pull request by askwang**
### Purpose

Add rollback to snapshot/tag/timestamp ddl syntax.
```sql
alter table t rollback to snapshot `2`;
alter table t rollback to tag `test-tag`;
alter table t rollback to timestamp `1730876906134`;
```

### CODE REVIEW

**Comment by LinMingQiang** [comment]
Why do we need a new `alter` ddl syntax. Is it the same as the `call`  procedure ?

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/RollbackExec.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            val snapshot = paimonTable.snapshotManager.earlierOrEqualTimeMills(version.toLong)
+            assert(snapshot != null)
+            paimonTable.rollbackTo(snapshot.id);
+
+          case _ =>
+            throw new UnsupportedOperationException(s"Unsupported rollback kind '$kind'.")
+        }
+      case t =>
+        throw new UnsupportedOperationException(
+          s"Can not delete tag for non-paimon FileStoreTable: $t")
```
Comment (wwj6591812) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/RollbackExec.scala:66]: Can not delete tag?

**Comment by askwang** [comment]
> Why do we need a new `alter` ddl syntax. Is it the same as the `call` procedure ?

SQL syntax support will be provided later, it is the same as procedure call.

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/RollbackExec.scala (hunk 1 above)
Comment (askwang) [reply paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/RollbackExec.scala]: my mistake, updated, ths

**Comment by LinMingQiang** [comment]
Why do we use  `alter table`  instead of `call` if they are the same?  And `alter table` should be used to modify the table metadata information, like alter table add col.

**Comment by askwang** [comment]
> Why do we use `alter table` instead of `call` if they are the same? And `alter table` should be used to modify the table metadata information, like alter table add col.

I think tag/snapshot are also part of the table meta info, maybe we can use alter table to modify it.

File: paimon-spark/paimon-spark-common/src/main/antlr4/org.apache.spark.sql.catalyst.parser.extensions/PaimonSqlExtensions.g4 [hunk 2]
```diff
@@ -104,6 +105,12 @@ timeUnit
   | MINUTES
   ;
 
+kind
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/antlr4/org.apache.spark.sql.catalyst.parser.extensions/PaimonSqlExtensions.g4:108]: timeTravelKind?

Comment (askwang) [reply paimon-spark/paimon-spark-common/src/main/antlr4/org.apache.spark.sql.catalyst.parser.extensions/PaimonSqlExtensions.g4]: changed.

**Comment by JingsongLi** [comment]
> Why do we use `alter table` instead of `call` if they are the same? And `alter table` should be used to modify the table metadata information, like alter table add col.

I agree with your idea. I have checked and currently no other lake formats have implemented custom syntax here. We can consider temporarily stopping and using CALL is not a bad idea.

@askwang What do you think?

**Comment by Zouxxyy** [comment]
In my mind, rollback should actually generate a new snapshot containing the changes, from this perspective, alter table should not be used. (although the current implementation directly deletes the existing snapshots).

Besides in spark doc description: ALTER TABLE statement changes the schema or properties of a table.

**Comment by askwang** [comment]
> > Why do we use `alter table` instead of `call` if they are the same? And `alter table` should be used to modify the table metadata information, like alter table add col.
> 
> I agree with your idea. I have checked and currently no other lake formats have implemented custom syntax here. We can consider temporarily stopping and using CALL is not a bad idea.
> 
> @askwang What do you think?

ok

**Pull request closed without merging**